nova_plugin/server.py

```diff
@@ -517,6 +521,45 @@ def _server_start(nova_client, server):
         ctx.logger.info('Server is already started?')
 
 
+@operation
+@with_nova_client
+def reboot(nova_client, reboot_type='soft', **kwargs):
+
+    if ctx.operation.retry_number == 0:
+        if reboot_type.upper() not in ['HARD', 'SOFT']:
+            raise NonRecoverableError(
+                'Unexpected reboot type: {}. '
+                'Valid values: SOFT or HARD.'.format(
+                    reboot_type))
+        server = get_server_by_context(nova_client)
+        nova_client.servers.reboot(server, reboot_type.upper())
+
+    server = nova_client.servers.get(server.id)
+
+    if server.status == SERVER_STATUS_REBOOT or \
+            server.status == SERVER_STATUS_HARD_REBOOT or \
+            server.status == SERVER_STATUS_UNKNOWN:
+        return ctx.operation.retry(
```

Comment:
What about using `server.status in [SERVER_STATUS_REBOOT,SERVER_STATUS_HARD_REBOOT, SERVER_STATUS_HARD_REBOOT]` would not be more readable ?